src/graph_node.rs

```diff
@@ -11,5 +11,20 @@ pub struct GraphNode<Id: TransactionId> {
     pub(crate) blocked_by_count: usize,
     /// Unique edges from this node.
     /// The number of edges is the same as the number of forks.
-    pub edges: AHashSet<Id>,
+    pub edges: Vec<Id>,
+}
+
+impl<Id: TransactionId> GraphNode<Id> {
+    /// Returns true if the edge was added successfully.
+    /// Edges are only added if the edge is unique.
+    /// Because we always add edges as we insert, the edge is unique if the
```

Comment:
Looks like a smart move to use `Vec`, just to confirm that `id`s are sorted in `Vec`? Without knowing the actual use cases, I am curious if this is a possible scenario: initially`vec![A, B, C]`, then `try_add_edge(B)`, which should return `false` in original version.

Reply:
Great question.
This is why I originally used a HashSet, I wanted a fast guarantee that the edge was unique.

I recently had the realization that the usage pattern is as follows:
1. insert_transaction
2. loop over all resource (account) accesses in the transaction, check for outstanding uses
3. For each conflicting outstanding use, we mutate the **blocking** transaction node's edges to add the `id` of the transaction we are inserting.

Since `prio_graph` is single-threaded, if transaction `2` conflicts with `1` on account `A` it will be added to the end of the `1`'s `edges`. If it again conflicts with `1` on account `B`, then `1`'s `edges` vec still has `2` at the end, since that was the most recent conflict.
Basically, because we always add edges for the conflicts in the logical order of our graph we can just check the most recent conflict on any blocking node.

Reply:
ah, single-threaded and logical order guarantees this behavior.